netutils/dhcpc/dhcpc.c

```diff
@@ -720,13 +724,18 @@ int dhcpc_request(FAR void *handle, FAR struct dhcpc_state *presult)
 
       /* Send the DISCOVER command */
 
-      ninfo("Broadcast DISCOVER\n");
-      if (dhcpc_sendmsg(pdhcpc, presult, DHCPDISCOVER) < 0)
+      if (last_send == 0 ||
```

Comment:
don't need changedhcp_send since it is impossible to block for long time..

Reply:
The change is to prevent dhcp_send from burst send multiple discover.
If not, every time we come back here we will send a dhcp. 
This is bad when we are receiving un-wanted dhcp packets, then the do-while will  keep sending

Reply:
In this case, we should loop in recv until the timeout

Reply:
According to RFC2131: "DHCP clients are responsible for all message retransmission.  The
   client MUST adopt a retransmission strategy that incorporates a
   randomized exponential backoff algorithm to determine the delay
   between retransmissions.  The delay between retransmissions SHOULD be
   chosen to allow sufficient time for replies from the server to be
   delivered based on the characteristics of the internetwork between
   the client and the server. "
It is better to let dhcp send delay be different from recv time out. So I think recv time out shouldn't be related to send  delay. That is the reason why I put a last_send here. 
In the future we might need to add an exponential backoff algorithm here

Reply:
I will add the exponential backoff now.

Reply:
remove exponential backoff